### PR TITLE
62 docs generation docs hotreload

### PR DESCRIPTION
**Pull request by TheAngryByrd**
## Proposed Changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to MiniScaffold?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Build and tests pass locally
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Content/Library/docsTool/Program.fs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        Paket.NuGetCache.GetTargetUserFolder package version </> "lib" </> rid
+
+
+    let copyAssets () =
+        Shell.copyDir (docsDir </> "content")   ( docsSrcDir </> "content") (fun _ -> true)
+        Shell.copyDir (docsDir </> "files")   ( docsSrcDir </> "files") (fun _ -> true)
+
+    let generateDocs (docSourcePaths : IGlobbingPattern) githubRepoName =
+        // This finds the current fsharp.core version of your solution to use for fsharp.literate
+        let fsharpCoreDir = locateDLL "FSharp.Core" "netstandard2.0"
```
Comment (baronfel) [inline Content/Library/docsTool/Program.fs]: One thing that could be done is to use dotnet-proj-info instead to get the resolved references of a project and find FSharp.Core from that set.  That would make this tool paket-agnostic.

File: Content/Library/docsTool/Program.fs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+        let parse (fileName : string) source =
+            let doc =
+                let fsharpCoreDir = sprintf "-I:%s" fsharpCoreDir
+                let runtimeDeps = "-r:System.Runtime -r:System.Net.WebClient"
```
Comment (baronfel) [inline Content/Library/docsTool/Program.fs]: For the set of dependencies to typecheck/run against, we may want to steal the logic from https://github.com/fsharp/FsAutoComplete/blob/master/src/FsAutoComplete.Core/FSIRefs.fs.  This improves on the assembly-location logic from FCS (which fails per https://github.com/dotnet/fsharp/issues/7701)

File: Content/Library/src/MyLib.1/Library.fs [hunk 3]
```diff
@@ -32,32 +29,16 @@ module Say =
...
-    /// **Output Type**
-    ///   * `string`
-    ///
-    /// **Exceptions**
-    ///
     [<CompiledName("Hiya")>]
     let hello name =
         sprintf "Hello %s" name
 
+    /// We did a bad api design
```
Comment (baronfel) [inline Content/Library/src/MyLib.1/Library.fs]: 🤣

**Comment by TheAngryByrd** [comment]
> This seems super-reasonable, and if you wanted to make the edit-templates-and-recompile cycle even better then you could even use fcswatch on the project. nice.
> 
> So a total-hotreload use case might be `fcswatch <project> -- watch <glob>`, right? To hotreload generation of the docs as you write them + hotreload restarting of your webserver when you change the templates?

Yeah that was the intention, though I was just using `dotnet watch` but same principal applies.

File: Content/Library/docsTool/Program.fs (hunk 1 above)
Comment (TheAngryByrd) [reply Content/Library/docsTool/Program.fs]: Great idea!

**Comment by dsyme** [comment]
Is this almost ready, and will it give a sample for .NET Core doc generation that I can copy into other repos? :-)

**Comment by TheAngryByrd** [comment]
Yeah this is pretty much good to go.  There's a few things that I don't have working such as:

- Generating a Nav structure from the folder structure
- Good styling / structure of pages
- Dealing with different Target Frameworks
- Dealing with multiple (older) versions of the documentation

But these aren't what I'd consider show stoppers.

**Comment by TheAngryByrd** [comment]
I'm gonna go ahead and merge this back into the other branch now, then clean it up more there.

**Pull request merged**